src/ok.rs

```diff
@@ -0,0 +1,125 @@
+    name: "CPU Manufacturer",
+    leaf: 0x00000000,
+    func: |res| {
+        let name: [u8; 12] = unsafe { transmute([res.ebx, res.edx, res.ecx]) };
+        let name = from_utf8(&name[..]).unwrap();
+
+        (name == "AuthenticAMD", None)
+    },
+}];
+
+const AMD_CPU_DEPENDENT_CPUIDS: &[CpuId] = &[
+    CpuId {
+        name: "Microcode support",
+        leaf: 0x80000002,
+        func: |_res| {
+            let cpu_name = {
+                let mut bytestr = Vec::with_capacity(48);
+                for cpuid in 0x8000_0002_u32..=0x8000_0004_u32 {
+                    let cpuid = unsafe { x86_64::__cpuid(cpuid) };
+                    let mut bytes: Vec<u8> = [cpuid.eax, cpuid.ebx, cpuid.ecx, cpuid.edx]
+                        .iter()
+                        .flat_map(|r| r.to_le_bytes().to_vec())
+                        .collect();
+                    bytestr.append(&mut bytes);
+                }
+                String::from_utf8(bytestr).unwrap().trim().to_string()
+            };
+
+            (cpu_name.to_uppercase().contains("EPYC"), None)
+        },
+    },
+    CpuId {
+        name: "Secure Memory Encryption (SME)",
+        leaf: 0x8000001f,
+        func: |res| (res.eax & 0x1 != 0, None),
+    },
+];
+
+const MICROCODE_SME_DEPENDENT_CPUIDS: &[CpuId] = &[CpuId {
+    name: "Secure Encrypted Virtualization (SEV)",
+    leaf: 0x8000001f,
+    func: |res| (res.eax & (0x1 << 1) != 0, None),
+}];
+
+const SEV_DEPENDENT_CPUIDS: &[CpuId] = &[
+    CpuId {
+        name: "Page flush MSR",
+        leaf: 0x8000001f,
+        func: |res| (res.eax & (0x1 << 2) != 0, None),
+    },
+    CpuId {
+        name: "Physical address bit reduction",
+        leaf: 0x8000001f,
+        func: |res| {
+            let field = res.ebx & 0b1111_1100_0000 >> 6;
+            (true, Some(format!("{}", field)))
+        },
+    },
+    CpuId {
+        name: "C-bit location in page table entry",
+        leaf: 0x8000001f,
+        func: |res| {
+            let field = res.ebx & 0b01_1111;
+            (true, Some(format!("{}", field)))
+        },
+    },
+    CpuId {
+        name: "Number of encrypted guests supported simultaneously",
+        leaf: 0x8000001f,
+        func: |res| (true, Some(format!("{}", res.ecx))),
+    },
+    CpuId {
+        name: "Minimum ASID value for SEV-enabled, SEV-ES disabled guest",
+        leaf: 0x8000001f,
+        func: |res| (true, Some(format!("{}", res.edx))),
+    },
+];
+
+// SEV-ES specific CPUIDs.
+const SEV_ES_CPUIDS: &[CpuId; 1] = &[CpuId {
+    name: "SEV Encrypted State (SEV-ES)",
+    leaf: 0x8000001f,
+    func: |res| (res.eax & (0x1 << 3) != 0, None),
+}];
+
+pub fn cmd(gen: Option<SevGeneration>, quiet: bool) -> Result<()> {
+    let mut passed = true;
+
+    if passed {
+        Ok(())
+    } else {
+        Err(error::Context::new(
+            "One or more tests in sevctl-ok reported a failure",
+            Box::<Error>::new(ErrorKind::InvalidData.into()),
+        ))
+    }
```

Comment:
For this commit I'd be tempted to just stub this with `todo!()` or something, but I get it's a stub and will change in a subsequent commit, so no big deal.